### PR TITLE
chore(ui-react): remove deprecated useAuthenticator internals

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
- remove `_state` usage specific updates
  - expose internal only `fields` on `useAuthenticator`
  - expose `getTotpSecretCode` on `useAuthenticator`
  - expose `socialProviders` on state machine facade
  - expose `unverifiedContactMethods` on state machine facade
    - replace `unverifiedAttributes` with `unverifiedContactMethods`

- remove `_send` usage specific updates
  - remove deprecation warning for `toFederatedSignIn` and use in RWA
  - expose `initializeMachine` on state machine facade

#### Issue #, if available
NA

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-core/src/Authenticator/hooks/useAuthenticator/__tests__/useAuthenticator.spec.tsx [hunk 1]
```diff
@@ -1,15 +1,53 @@
...
 
 import { AuthenticatorProvider } from '../../../context';
 import { USE_AUTHENTICATOR_ERROR } from '../constants';
+import * as utils from '../utils';
 import { useAuthenticator, UseAuthenticator } from '..';
 
+const mockServiceFacade: AuthenticatorServiceFacade = {
+  authStatus: 'authenticated',
+  codeDeliveryDetails: {} as UseAuthenticator['codeDeliveryDetails'],
+  error: undefined as unknown as UseAuthenticator['error'],
```
Comment (joebuono) [inline packages/react-core/src/Authenticator/hooks/useAuthenticator/__tests__/useAuthenticator.spec.tsx:14]: Just curious if there's a cleaner way to cast this

File: packages/react-core/src/Authenticator/hooks/useAuthenticator/utils.ts [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+
+import { COMPONENT_ROUTE_KEYS } from './constants';
+import {
+  AuthenticatorRouteComponentKey,
+  AuthenticatorLegacyFields,
+  Comparator,
+  Selector,
+} from './types';
+
+export const defaultComparator = (): false => false;
```
Comment (joebuono) [inline packages/react-core/src/Authenticator/hooks/useAuthenticator/utils.ts:20]: What does this do?

File: packages/ui/src/helpers/authenticator/__tests__/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { getTotpCodeURL } from '../utils';
+
+const SECRET_KEY = 'shhhhh';
```
Comment (joebuono) [inline packages/ui/src/helpers/authenticator/__tests__/utils.test.ts:3]: 😂

File: packages/ui/src/helpers/authenticator/facade.ts [hunk 4]
```diff
@@ -112,20 +118,24 @@ export const getSendEventAliases = (
 export const getServiceContextFacade = (
   state: AuthMachineState
 ): AuthenticatorServiceContextFacade => {
-  const actorState = getActorState(state);
   const actorContext = (getActorContext(state) ?? {}) as ActorContextWithForms;
   const {
     codeDeliveryDetails,
     remoteError: error,
     validationError: validationErrors,
+    unverifiedContactMethods,
```
Comment (joebuono) [inline packages/ui/src/helpers/authenticator/facade.ts:126]: nit: alphabetize

**Review by joebuono — APPROVED** [review]
Thanks for doing this! Just had a few small questions/nits

File: packages/react-core/src/Authenticator/hooks/useAuthenticator/__tests__/useAuthenticator.spec.tsx (hunk 1 above)
Comment (calebpollman) [reply packages/react-core/src/Authenticator/hooks/useAuthenticator/__tests__/useAuthenticator.spec.tsx:14]: Not really - `AuthenticatorServiceFacade` requires `error` field to be of type `string` so the cast to `unknown` is required unless we update the `AuthenticatorServiceFacade` type

File: packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts [hunk 5]
```diff
@@ -67,7 +67,7 @@ export class AuthenticatorComponent
       formFields,
     } = this;
 
-    const { authService } = this.authenticator;
+    const { authService, initializeMachine } = this.authenticator;
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/components/authenticator/components/authenticator/authenticator.component.ts:70]: Welcome to Angular 😈

File: packages/vue/src/components/verify-user.vue [hunk 6]
```diff
@@ -25,7 +25,7 @@ const actorState: ComputedRef<SignInState> = computed(
   () => getActorState(state.value) as SignInState
 );
 
-const unverifiedAttributes = actorState.value.context.unverifiedAttributes;
+const unverifiedContactMethods = actorState.value.context.unverifiedContactMethods;
```
Comment (wlee221) [inline packages/vue/src/components/verify-user.vue:28]: Weird nit, but we can destructure this

**Review by wlee221 — APPROVED** [review]
Nothing blocking, lgtm

**Review by wlee221 — APPROVED** [review]

**Review by joebuono — APPROVED** [review]

**Pull request merged**